apps/www/components/ui/navigation-menu.tsx

```diff
@@ -0,0 +1,122 @@
+  <NavigationMenuPrimitive.Trigger
+    className={cn(
+      navigationMenuTriggerStyle(),
+      "group flex items-center justify-center gap-0.5",
+      className
+    )}
+    {...props}
+  >
+    {children}{" "}
+    <ChevronDown
+      className="relative top-[1px] h-4 w-4 transition duration-200 group-data-[state=open]:rotate-180"
+      aria-hidden
+    />
+  </NavigationMenuPrimitive.Trigger>
+)
+
+const navigationMenuLinkVariants = cva([], {
+  variants: {
+    variant: {
+      default: cn(navigationMenuTriggerStyle(), "block leading-none"),
+      unstyled: "",
+    },
+  },
+  defaultVariants: {
+    variant: "default",
+  },
+})
+
+export const NavigationMenuLink: React.FC<
+  NavigationMenuPrimitive.NavigationMenuLinkProps &
+    VariantProps<typeof navigationMenuLinkVariants>
+> = ({ className, variant, ...props }) => (
+  <NavigationMenuPrimitive.Link
+    className={cn(navigationMenuLinkVariants({ variant, className }))}
+    {...props}
+  />
+)
+
+export const NavigationMenuContent: React.FC<
+  NavigationMenuPrimitive.NavigationMenuContentProps
+> = ({ className, ...props }) => (
+  <NavigationMenuPrimitive.Content
+    className={cn(
+      [
+        "absolute top-0 left-0 w-full duration-300 md:w-auto",
+        "data-[motion^=from-]:animate-in data-[motion^=to-]:animate-out data-[motion^=from-]:fade-in data-[motion^=to-]:fade-out data-[motion=to-start]:slide-out-to-left-52 data-[motion=to-end]:slide-out-to-right-52 data-[motion=from-start]:slide-in-from-left-52 data-[motion=from-end]:slide-in-from-right-52",
+      ],
+      className
+    )}
+    {...props}
+  />
+)
+
+export const NavigationMenuIndicator: React.FC<
+  NavigationMenuPrimitive.NavigationMenuIndicatorProps
+> = ({ className, ...props }) => (
+  <NavigationMenuPrimitive.Indicator
+    className={cn(
+      "top-full z-[1] flex h-1.5 items-end justify-center overflow-hidden data-[state=visible]:animate-in data-[state=hidden]:animate-out data-[state=visible]:fade-in data-[state=hidden]:fade-out",
+      className
+    )}
+    {...props}
+  >
+    <div className="relative top-[60%] h-1.5 w-1.5 rotate-45 rounded-tl-sm bg-slate-200 dark:bg-slate-700" />
+  </NavigationMenuPrimitive.Indicator>
+)
+
+export const NavigationMenuViewport: React.FC<
+  NavigationMenuPrimitive.NavigationMenuViewportProps
+> = ({ className, ...props }) => (
+  <div
+    className={cn("absolute left-0 top-full flex w-full justify-center")}
+    id="viewport"
+  >
+    <NavigationMenuPrimitive.Viewport
+      className={cn(
+        "origin-top-center relative mt-1.5 h-[var(--radix-navigation-menu-viewport-height)] w-full overflow-hidden rounded-md border border-slate-200 bg-white shadow-lg data-[state=open]:animate-in data-[state=open]:zoom-in dark:border-slate-700 dark:bg-slate-800 md:w-[var(--radix-navigation-menu-viewport-width)]",
+        className
+      )}
+      {...props}
+    />
+  </div>
+)
+
+export const NavigationMenuSub = NavigationMenuPrimitive.Sub
```

Comment:
Let's also move all export at the end of the file. See

https://github.com/shadcn/ui/blob/6ea6856f5a1082d4d9c231559b6bc3ee73827493/apps/www/components/ui/dropdown-menu.tsx#L187-L203

I like inline export too but I found it a bit easier to manage when we have all export grouped together in the case of primitives.